src/app/modules/submission/email-submission/email-submission.util.ts

```diff
@@ -518,3 +519,31 @@ export const concatAttachmentsAndResponses = (
   response += attachments.reduce((acc, { content }) => acc + content, '')
   return response
 }
+
+// Class to encapsulate maskedData method
+// which returns autoReplyData with masked Corppass UID
+
+export class autoReplyDataObj {
+  data: EmailAutoReplyField[]
+  constructor(autoReplyData: EmailAutoReplyField[]) {
+    this.data = autoReplyData
+  }
+  get maskedData(): EmailAutoReplyField[] {
+    const maskField = (field: EmailAutoReplyField) => {
```

Comment:
this function should be a class member, rather than a new function created every single time `maskedData` is retrieved

Reply:
edited